webrtc/interfaces.html

```diff
@@ -0,0 +1,74 @@
+<!doctype html>
+<html>
+  <head>
+    <title>"WebRTC IDL Tests</title>
+  </head>
+  <body>
+    <script src=/resources/testharness.js></script>
+    <script src=/resources/testharnessreport.js></script>
+    <script src=/resources/WebIDLParser.js></script>
+    <script src=/resources/idlharness.js></script>
+    <script>
+      'use strict';
+
+      function initRTCCertificate(idlArray) {
+        return RTCPeerConnection.generateCertificate({
+          name: 'RSASSA-PKCS1-v1_5',
+          modulusLength: 2048,
+          publicExponent: new Uint8Array([1, 0, 1]),
+          hash: 'SHA-256'
+        })
+        .then(cert => {
+          window.rtcCert = cert;
+          idlArray.add_objects({'RTCCertificate': ['rtcCert']});
+         });
+      }
+
+      // A customized version of test_harness that returns
+      // promise of the test completing as well as allow
+      // other promise tests to run in parallel.
+      function promise_test(testFunc, testName, properties) {
```

Comment:
The problem is that the original `promise_test()` cannot be used here, because it is nested inside another promise_test. And promise_test only execute a test after the previous test promise is resolved, thus causing deadlock and time out the entire test.

Technically I can inline the code inside `initRTCCertificate()`. But I feel that it is clearer as a separate function and more async IDL tests will be added here anyway.

I tentatively named it `promise_test` to serve as discussion on whether it makes any sense and whether something like this is worth to be added into testharness.js.